cli/main.go

```diff
@@ -173,6 +174,12 @@ func (c *DiceDBClient) handleWatchCommand(cmd string, args []string) {
 	go c.watchCommand(baseCmd, toArgInterface(args[1:])...)
 }
 
+func (c *DiceDBClient) handleWatchModeExit() {
+	c.subCancel()
+	c.subscribed = false
+	c.subType = ""
```

Comment:
Will this be redundant since we already do this as part of `watchCommand()` defer?